cloud/linode/loadbalancers_test.go

```diff
@@ -443,6 +448,198 @@ func testCreateNodeBalancerWithInvalidFirewall(t *testing.T, client *linodego.Cl
 	}
 }
 
+func testUpdateLoadBalancerAddNode(t *testing.T, client *linodego.Client, f *fakeAPI) {
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: randString(),
+			UID:  "foobar1234",
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name:     randString(),
+					Protocol: "TCP",
+					Port:     int32(80),
+					NodePort: int32(30000),
+				},
+			},
+		},
+	}
+
+	nodes1 := []*v1.Node{
+		{
+			Status: v1.NodeStatus{
+				Addresses: []v1.NodeAddress{
+					{
+						Type:    v1.NodeInternalIP,
+						Address: "127.0.0.1",
+					},
+				},
+			},
+		},
+	}
+
+	nodes2 := []*v1.Node{
+		{
+			Status: v1.NodeStatus{
+				Addresses: []v1.NodeAddress{
+					{
+						Type:    v1.NodeInternalIP,
+						Address: "127.0.0.1",
+					},
+				},
+			},
+		},
+		{
+			Status: v1.NodeStatus{
+				Addresses: []v1.NodeAddress{
+					{
+						Type:    v1.NodeInternalIP,
+						Address: "127.0.0.2",
+					},
+				},
+			},
+		},
+		{
+			Status: v1.NodeStatus{
+				Addresses: []v1.NodeAddress{
+					{
+						Type:    v1.NodeInternalIP,
+						Address: "127.0.0.3",
+					},
+				},
+			},
+		},
+	}
+
+	lb := &loadbalancers{client, "us-west", nil}
+	fakeClientset := fake.NewSimpleClientset()
+	lb.kubeClient = fakeClientset
+
+	defer func() {
+		_ = lb.EnsureLoadBalancerDeleted(context.TODO(), "linodelb", svc)
+	}()
+
+	lbStatus, err := lb.EnsureLoadBalancer(context.TODO(), "linodelb", svc, nodes1)
+	if err != nil {
+		t.Errorf("EnsureLoadBalancer returned an error %s", err)
+	}
+	svc.Status.LoadBalancer = *lbStatus
+
+	stubService(fakeClientset, svc)
+
+	f.ResetRequests()
+
+	err = lb.UpdateLoadBalancer(context.TODO(), "linodelb", svc, nodes1)
+	if err != nil {
+		t.Errorf("UpdateLoadBalancer returned an error while updated LB to have one node: %s", err)
+	}
+
+	checkIDs := func() (int, int) {
+		rx := regexp.MustCompile("/nodebalancers/[0-9]+/configs/[0-9]+/rebuild")
+
+		var req *fakeRequest
+		for request := range f.requests {
+			request := request
+			if rx.MatchString(request.Path) {
+				req = &request
+				break
+			}
+		}
+
+		if req == nil {
+			t.Fatalf("Nodebalancer config rebuild request was not called.")
+		}
+
+		var nbcro linodego.NodeBalancerConfigRebuildOptions
+
+		err = json.Unmarshal([]byte(req.Body), &nbcro)
+		if err != nil {
+			t.Fatalf("Unable to unmarshall request body %#v, error: %#v", req.Body, err)
+		}
```

Comment:
```suggestion
		if err := json.Unmarshal([]byte(req.Body), &nbcro); err != nil {
			t.Fatalf("Unable to unmarshall request body %#v, error: %#v", req.Body, err)
		}
```

nit: More go-y to inline the err check